test/utils/utils.test.js

```diff
@@ -0,0 +1,200 @@
+    });
+    it('should set parent innerHTML on success', async () => {
+      const svg = { src: '/foo.svg', parentElement: { innerHTML: '' } };
+      fetchStub.resolves({ ok: true, text: () => Promise.resolve('<svg></svg>') });
+      await utils.loadSvgs([svg]);
+      expect(svg.parentElement.innerHTML).to.equal('<svg></svg>');
+    });
+    it('should remove svg on error', async () => {
+      const svg = { src: '/foo.svg', parentElement: {}, remove: sinon.stub() };
+      fetchStub.rejects(new Error('fail'));
+      await utils.loadSvgs([svg]);
+      expect(svg.remove.called).to.be.true;
+    });
+  });
+
+  describe('createActionBtn', () => {
+    it('should create a button with text', async () => {
+      const btnCfg = document.createElement('div');
+      btnCfg.innerText = 'Test Button';
+      const btn = await utils.createActionBtn(btnCfg, 'test-class');
+      expect(btn).to.be.instanceOf(HTMLElement);
+      expect(btn.className).to.include('test-class');
+      expect(btn.textContent).to.include('Test Button');
+    });
+    it('should prepend text if swapOrder is true', async () => {
+      const btnCfg = document.createElement('div');
+      btnCfg.innerText = 'Test Button';
+      const btn = await utils.createActionBtn(btnCfg, 'test-class', false, true);
+      expect(btn.firstChild.className).to.include('btn-text');
+    });
+  });
+
+  describe('showErrorToast', () => {
+    let fetchStub;
+    let origImport;
+    beforeEach(() => {
+      fetchStub = sinon.stub(window, 'fetch');
+      fetchStub.onFirstCall().resolves({ text: () => Promise.resolve('<svg>alert</svg>') });
+      fetchStub.onSecondCall().resolves({ text: () => Promise.resolve('<svg>close</svg>') });
+      origImport = window.import;
+      window.import = () => Promise.resolve({ decorateDefaultLinkAnalytics: sinon.stub() });
+    });
+    afterEach(() => {
+      fetchStub.restore();
+      window.import = origImport;
+    });
+    it('should append error toast if not present and set message', async () => {
+      const targetEl = document.createElement('div');
+      const unityEl = document.createElement('div');
+      const msgEl = document.createElement('span');
+      msgEl.textContent = 'Error message';
+      unityEl.appendChild(document.createElement('div'));
+      unityEl.appendChild(msgEl);
+      unityEl.querySelector = () => unityEl.children[0];
+      document.body.innerHTML = '<div class="unity-enabled"><div class="interactive-area"><div class="alert-holder"><div class="alert-toast"><div class="alert-text"><p></p></div></div></div></div></div>';
+      await utils.showErrorToast(targetEl, unityEl, 'div');
+      const p = document.querySelector('.unity-enabled .interactive-area .alert-holder .alert-toast .alert-text p');
+      expect(p.innerText).to.equal('Error message');
+      const holder = document.querySelector('.unity-enabled .interactive-area .alert-holder');
+      expect(holder.style.display).to.equal('flex');
+    });
+  });
```

Comment:
Just adding a note that these will most likely need to be removed later once these Ps-specific functions are removed from our utils file.